### PR TITLE
Feature/datasets get all

### PR DESCRIPTION
**Pull request by jamie-short**
Adding in get command for:
- get datasets

At this point there are no options, or filters applied, this qill be added in a subsequent pr

I have refactored the api code, to split out the routes for models and datasets individually as tis starts to grow

### CODE REVIEW

File: test/api/test_dafni_api.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        mock_request.get.return_value = request_response_fixture
+
+        # setup data for call
+        url = "dafni/models/url"
+        jwt = JWT
+
+        # CALL
+        # ASSERT
+        with pytest.raises(HTTPError, match="404 client model"):
+            dafni_api.dafni_get_request(url, jwt)
```
Comment (iroper-tessella) [inline test/api/test_dafni_api.py:54]: Tests for the post requests? I imagine they will be fairly similar.

File: test/api/test_dataset_api.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import pytest
```
Comment (iroper-tessella) [inline test/api/test_dataset_api.py:1]: Small thing, but the file is called "dataset_api.py", as opposed to "datasets_api.py".

File: test/fixtures/dataset_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import pytest
+from typing import List
+
+
+@pytest.fixture
+def get_dataset_list_fixture() -> List[dict]:
+    """Test fixture for simulating the datset data return
```
Comment (iroper-tessella) [inline test/fixtures/dataset_fixtures.py:7]: typo in dataset

File: test/test_get.py [hunk 4]
```diff
@@ -281,3 +283,64 @@ def test_message_printed_when_no_id_provided(
...
+            self,
+            mock_create,
+            mock_jwt,
+            mock_get,
+            mock_output,
+            processed_jwt_fixture,
+            get_dataset_list_fixture,
+        ):
+            # SETUP
+            # setup get group command
```
Comment (iroper-tessella) [inline test/test_get.py:304]: I don't think this comment describes what happens below it

**Review by iroper-tessella — CHANGES_REQUESTED** [review]
Looks great, just a few comments of things to be added and reviewed.

File: dafni_cli/api/dafni_api.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    Args:
+        url (str): The url endpoint that is being queried
+        jwt (str): JWT
+        data (dict): The data to be POSTed in JSON format
+        allow_redirect (bool): Flag to allow redirects during API call. Defaults to False.
+
+    Returns:
+        list: For an endpoint returning several objects, a list is returned (e.g. /models/).
+        dict: For an endpoint returning one object, this will be a dictionary (e.g. /models/<version_id>).
```
[the remark this answers is not in the document]
Comment (jamie-short) [reply dafni_cli/api/dafni_api.py:45]: good point, updated

File: dafni_cli/datasets/dataset.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+from typing import List
+
+from dafni_cli.consts import TAB_SPACE, CONSOLE_WIDTH
+from dafni_cli.model import prose_print
+
+
+class Dataset:
+    """Class to represent the DAFNI Dataset
+    client model
+    """
```
[the remark this answers is not in the document]
Comment (jamie-short) [reply dafni_cli/datasets/dataset.py:12]: updated

File: test/api/test_dafni_api.py (hunk 1 above)
Comment (jamie-short) [reply test/api/test_dafni_api.py:54]: good spot, tests added

File: test/api/test_dataset_api.py (hunk 2 above)
Comment (jamie-short) [reply test/api/test_dataset_api.py:1]: updated, and for model

File: test/fixtures/dataset_fixtures.py (hunk 3 above)
Comment (jamie-short) [reply test/fixtures/dataset_fixtures.py:7]: fixed

File: test/test_get.py (hunk 4 above)
Comment (jamie-short) [reply test/test_get.py:304]: updated

File: test/api/test_dafni_api.py [hunk 7]
```diff
@@ -52,3 +52,77 @@ def test_exception_raised_for_failed_call(
         # ASSERT
         with pytest.raises(HTTPError, match="404 client model"):
             dafni_api.dafni_get_request(url, jwt)
+
+
+@patch("dafni_cli.api.dafni_api.requests")
+class TestDafniGPostRequest:
```
Comment (iroper-tessella) [inline test/api/test_dafni_api.py:58]: Typo in name of class

**Review by iroper-tessella — CHANGES_REQUESTED** [review]
Just one typo and then I am happy to merge.

**Review by iroper-tessella — APPROVED** [review]
Looks good. Approved.

**Pull request merged**